### PR TITLE
Fix mixup between httplib and urllib3

### PR DESCRIPTION
**Pull request by lmazuel**
Trying to fix https://github.com/kevin1024/vcrpy/issues/315, I found this, that I hoped would have fixed the bug, but alas it just moves it a little further (so I'm still struggling to find the root cause of #315)

At least, even if it's not enough for #315, I think this is still a typo in the code that should be fixed.

My two cents,

### CODE REVIEW

**Comment by graingert** [comment]
Can you provide a test for this?

**Comment by lmazuel** [comment]
I honestly have no idea how to test vcrpy itself, I got this by scrutiny...

**Comment by lmazuel** [comment]
@graingert It seems `reset_patchers` is not tested at all, or I don't understand how it's tested.

**Comment by lmazuel** [comment]
@graingert I pushed a test, that I confirmed breaks in `pyXX-urllib3121` before the patch I propose, but work after.
I still fight with `pyXX-requests216` that for some reason fails on Travis for test_urllib3.

**Comment by lmazuel** [comment]
@graingert Travis is green, youhou! So this PR fixes actually two bugs. 

Little summary:
- `reset_patchers` that is used in `force_reset` was resetting incorrectly urllib3 `HTTPConnection` class to `httlib.HTTPConnection` and not `urllib3.connection.HTTPConnection`. Easy part.
- Starting `requests` 2.16.3, [requests.packages](https://github.com/requests/requests/blob/master/requests/packages.py) is a black magic module that says `requests.packages.urllib3` == `urllib3`. This becomes an actual namespace mess for `mock.patch`, when the code was trying to unmock `requests.packages.urllib3.connection.HTTPConnection` **AND** `urllib3.connection.HTTPConnection`. Meaning that `force_reset` was still not returning the correct HTTPConnection class, but the fake one from VCR this time. So this PR also ask `force_reset` to do not try to unmock if `requests.packages` is the black magic file.
- I added a unittest and requests 2.18.4 to Travis just in case

File: vcr/patch.py [hunk 1]
```diff
@@ -362,8 +362,22 @@ def __exit__(self, *args):
...
     yield mock.patch.object(httplib, 'HTTPConnection', _HTTPConnection)
     yield mock.patch.object(httplib, 'HTTPSConnection', _HTTPSConnection)
+
     try:
-        import requests.packages.urllib3.connectionpool as cpool
+        import requests
+        if requests.__build__ < 0x021603:
+            # Avoid double unmock if requests 2.16.3
+            # First, this is pointless, requests.packages.urllib3 *IS* urllib3 (see package.py)
+            # Second, this unmock twice the same thing with different namespace
```
Comment (graingert) [inline vcr/patch.py]: grammar

**Comment by graingert** [comment]
nice, can you rebase and clean up your commits.

**Comment by kevin1024** [comment]
Haha hacks causing hacks all the way up the stack.  Thanks for this, hopefully will solve a lot of the issues people are seeing.

**Comment by IvanMalison** [comment]
> nice, can you rebase and clean up your commits.

I'd prefer to keep all of these commits, since they aren't doing different things.

**Comment by IvanMalison** [comment]
@lmazuel can you provide the new stack trace that youa re getting?

**Comment by graingert** [comment]
@IvanMalison those commits were post cleanup.

**Comment by lmazuel** [comment]
@IvanMalison I like atomic commit as well, so I rebased 6 commits to 3 (do some fixup on commit named like "typo", etc.)

**Pull request merged**